File: fix-pm2.js
const { Client } = require('ssh2');
const HOST = '120.79.17.184';
const USER = 'root';
const PASS = 'Wangjie910621';

const commands = [
  `echo "===== PM2 修复 ====="`,
  // 先停掉当前占用 3001 的进程
  `kill -9 $(lsof -t -i:3001) 2>/dev/null; echo "killed old process"`,
  `sleep 1`,
  // 重启 PM2
  `cd /home/admin/ai-nexus && pm2 start server.js --name nexus-hub 2>&1 | tail -10`,
  `pm2 save 2>&1`,
  `pm2 list 2>&1`,
  
  `echo "===== 后端 3001 根路径测试 ====="`,
  `sleep 2`,
  `curl -s -o /dev/null -w "localhost:3001/ -> %{http_code}\\n" http://localhost:3001/`,
  `curl -s -o /dev/null -w "localhost:3001/studio/ -> %{http_code}\\n" http://localhost:3001/studio/`,
  `curl -s -o /dev/null -w "localhost:3001/api/ -> %{http_code}\\n" http://localhost:3001/api/`,
  
  `echo "===== 后端日志 ====="`,
  `pm2 logs nexus-hub --lines 15 --nostream 2>&1 | tail -20`,
  
  `echo "===== 服务器信息 ====="`,
  `ip addr show | grep "inet " | grep -v 127.0.0.1`,
  `curl -s ifconfig.me 2>&1`,
];

const conn = new Client();
conn.on('ready', () => { runCommands(0); });
conn.on('error', (err) => { console.error('Error:', err.message); process.exit(1); });
function runCommands(idx) {
  if (idx >= commands.length) { conn.end(); return; }
  conn.exec(commands[idx], (err, stream) => {
    if (err) { console.log('ERR:', err.message); runCommands(idx + 1); return; }
    let out = '';
    stream.on('data', d => out += d.toString());
    stream.on('stderr', d => out += d.toString());
    stream.on('close', () => { if (out.trim()) console.log(out.trim()); runCommands(idx + 1); });
  });
}
conn.connect({ host: HOST, port: 22, username: USER, password: PASS });

File: download/fix_pm2.py
import paramiko
import sys

host = "120.79.17.184"
port = 22
username = "root"
password = os.environ.get("DEPLOY_PASS", "CHANGE_ME")

ssh = paramiko.SSHClient()
ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())

try:
    print("Connecting...")
    ssh.connect(host, port, username, password, look_for_keys=False, allow_agent=False)
    print("Connected!")
    
    def run(cmd):
        stdin, stdout, stderr = ssh.exec_command(cmd)
        out = stdout.read().decode().strip()
        err = stderr.read().decode().strip()
        if out:
            print(out)
        if err:
            print(f"STDERR: {err}")
        return out, err
    
    # Check PM2 logs for nexus-hub
    print("=== PM2 logs (last 50 lines) ===")
    run("pm2 logs nexus-hub --lines 50 --nostream")
    
    print("=== Checking server.js exists ===")
    run("ls -la /home/admin/ai-nexus/server.js")
    
    print("=== Checking Node.js version ===")
    run("node -v")
    
    print("=== Checking if port 3001 is in use ===")
    run("ss -tlnp | grep 3001")
    
    print("=== Try restarting PM2 ===")
    run("pm2 delete nexus-hub")
    run("cd /home/admin/ai-nexus && pm2 start server.js --name nexus-hub 2>&1")
    run("sleep 2 && pm2 list")
    
    print("=== PM2 logs after restart ===")
    run("pm2 logs nexus-hub --lines 30 --nostream")
    
    print("Done!")
    
except Exception as e:
    print(f"Error: {e}")
    sys.exit(1)
finally:
    ssh.close()
